cirq/work/observable_measurement_test.py

```diff
@@ -213,31 +260,73 @@ def test_meas_specs_still_todo():
     bsa = cw.BitstringAccumulator(
         meas_spec, [], {q: i for i, q in enumerate(cirq.LineQubit.range(3))}
     )
+    return bsa, meas_spec
+
+
+def test_meas_specs_still_todo():
+    bsa, meas_spec = _set_up_meas_specs_for_testing()
+    stop = cw.RepetitionsStoppingCriteria(1_000)
 
     # 1. before taking any data
     still_todo, reps = _check_meas_specs_still_todo(
-        meas_specs=[meas_spec], accumulators={meas_spec: bsa}, desired_repetitions=1_000
+        meas_specs=[meas_spec],
+        accumulators={meas_spec: bsa},
+        stopping_criteria=stop,
     )
     assert still_todo == [meas_spec]
     assert reps == 1_000
 
     # 2. After taking a mocked-out 997 shots.
     bsa.consume_results(np.zeros((997, 3), dtype=np.uint8))
     still_todo, reps = _check_meas_specs_still_todo(
-        meas_specs=[meas_spec], accumulators={meas_spec: bsa}, desired_repetitions=1_000
+        meas_specs=[meas_spec],
+        accumulators={meas_spec: bsa},
+        stopping_criteria=stop,
     )
     assert still_todo == [meas_spec]
     assert reps == 3
 
     # 3. After taking the final 3 shots
     bsa.consume_results(np.zeros((reps, 3), dtype=np.uint8))
     still_todo, reps = _check_meas_specs_still_todo(
-        meas_specs=[meas_spec], accumulators={meas_spec: bsa}, desired_repetitions=1_000
+        meas_specs=[meas_spec],
+        accumulators={meas_spec: bsa},
+        stopping_criteria=stop,
     )
     assert still_todo == []
     assert reps == 0
 
 
+def test_meas_spec_still_todo_bad_spec():
+    bsa, meas_spec = _set_up_meas_specs_for_testing()
+
+    class BadStopping(StoppingCriteria):
+        def more_repetitions(self, accumulator: BitstringAccumulator) -> int:
+            return -23
+
+    bad_stop = BadStopping()
+    with pytest.raises(ValueError, match='positive'):
+        _, _ = _check_meas_specs_still_todo(
+            meas_specs=[meas_spec],
+            accumulators={meas_spec: bsa},
+            stopping_criteria=bad_stop,
+        )
+
+
+def test_meas_spec_still_todo_lots_of_params():
+    # This test is fairly slow (1.5s) because we need at least 300_001 measurement specs
+    # to trigger the exception being tested.
```

Comment:
You should be able to mock out MAX_REPETITIONS_PER_JOB to speed this up ([SO example](https://stackoverflow.com/questions/38336090/mocking-a-global-variable)):
```
from unittest import mock

mock_max_reps = mock.patch.object(cw, 'MAX_REPETITIONS_PER_JOB', 1000)
with mock_max_reps:
    # ...run test...
```

Reply:
cool; I used pytest's monkeypatch fixture which is short and cleans itself up afterwards